### PR TITLE
build: inform user that packing charm may take time (CRAFT-347)

### PR DESCRIPTION
**Pull request by cjp256**
Let the user know which charm is being packed and that it may take some
time.  Move launched_environment() logic into pack_charm_in_instance()
to have the information required (charm name) prior to launching.

Example output:

```
$ charmcraft pack
Packing charm 'foo_ubuntu-18.04-amd64.charm' which may take several minutes...
Created 'foo_ubuntu-18.04-amd64.charm'.
Packing charm 'foo_ubuntu-20.04-amd64.charm' which may take several minutes...
Created 'foo_ubuntu-20.04-amd64.charm'.
```

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

### CODE REVIEW

File: charmcraft/commands/build.py [hunk 1]
```diff
@@ -265,17 +261,25 @@ def pack_charm_in_instance(self, instance: Executor, bases_index: int) -> str:
...
-                cmd,
-                check=True,
-                cwd=get_managed_environment_project_path().as_posix(),
-            )
-        except subprocess.CalledProcessError as error:
-            capture_logs_from_instance(instance)
-            raise CommandError(
-                f"Failed to build charm for bases index '{bases_index}'."
-            ) from error
+        logger.info(f"Packing charm {charm_name!r} which may take several minutes...")
```
Comment (sergiusens) [inline charmcraft/commands/build.py:264]: Mentioning minutes gives me a Windows 95 progress bar vibe, won't age well
```suggestion
        logger.info(f"Packing charm {charm_name!r}...")
```

Comment (cjp256) [reply charmcraft/commands/build.py:264]: Haha, well I expect it to be redone in a month or so with @facundos new messages / status handling.  Will update.

Comment (cjp256) [reply charmcraft/commands/build.py]: Done

**Review by sergiusens — APPROVED** [review]

**Review by facundobatista — APPROVED** [review]
+1

**Pull request merged**